### PR TITLE
fix dead links to botorch.[generation].gen (generation part added)

### PR DESCRIPTION
**Pull request by sgbaird**
## Motivation

Some dead links exist on the docs page.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

For example:
https://botorch.org/api/gen.html#botorch.gen.gen_candidates_scipy is a dead link
https://botorch.org/api/gen.html#botorch.generation.gen.gen_candidates_scipy is the correct link

## Related PRs

N/A

### CODE REVIEW

File: docs/optimization.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ observations.
 #### Using scipy Optimizers on Tensors
 
 The default method used by BoTorch to optimize acquisition functions is
-[`gen_candidates_scipy()`](../api/gen.html#botorch.gen.gen_candidates_scipy).
+[`gen_candidates_scipy()`](../api/gen.html#botorch.generation.gen.gen_candidates_scipy).
```
Comment (Balandat) [inline docs/optimization.md:20]:
```suggestion
[`gen_candidates_scipy()`](../api/generation.html#botorch.generation.gen.gen_candidates_scipy).
```

File: docs/optimization.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ used directly, without the need to perform `numpy` conversion. These first-order
 gradient-based optimizers are particularly useful for the case when the
 acquisition function is stochastic, where algorithms like L-BFGS or SLSQP that
 are designed for deterministic functions should not be applied. The function
-[`gen_candidates_torch()`](../api/gen.html#botorch.gen.gen_candidates_torch)
+[`gen_candidates_torch()`](../api/gen.html#botorch.generation.gen.gen_candidates_torch)
```
Comment (Balandat) [inline docs/optimization.md:35]:
```suggestion
[`gen_candidates_torch()`](../api/generation.html#botorch.generation.gen.gen_candidates_torch)
```

File: docs/optimization.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ defined as the sum of the $b$ individual q-batch acquisition values.
 The wrapper function
 [`optimize_acqf()`](../api/optim.html#botorch.optim.optimize.optimize_acqf)
 uses
-[`get_best_candidates()`](../api/gen.html#botorch.gen.get_best_candidates)
+[`get_best_candidates()`](../api/gen.html#botorch.generation.gen.get_best_candidates)
```
Comment (Balandat) [inline docs/optimization.md:62]:
```suggestion
[`get_best_candidates()`](../api/generation.html#botorch.generation.gen.get_best_candidates)
```

**Review by Balandat — CHANGES_REQUESTED** [review]
Thanks for catching these - I guess you also need to update the base url here though.

**Comment by sgbaird** [comment]
Sure thing! And good follow-up catch

**Review by Balandat — APPROVED** [review]
thanks!

**Pull request closed without merging**